wp-dependency-installer.php

```diff
@@ -312,6 +317,9 @@ public function ajax_router() {
 		 * @return boolean True if required. Default: False
 		 */
 		public function is_required( &$plugin ) {
+			if ( empty( $this->config ) ) {
+				return false;
+			}
```

Comment:
This similar check is in `run()` and so shouldn't be needed here.

Reply:
and what if for some strange reason this sequence happens:

```php
WP_Dependency_Installer::instance()->is_required( "github-updater/github-updater.php" );

...

WP_Dependency_Installer::instance()->run( __DIR__ );

```

Reply:
There is no way to account for every possibility. What do you think that example would do? There is no URL to even try to install.

Reply:
I think this way is more explicit way to read function code ( a dependency cannot be `required`, if a configuration file doesn't exist yet )

BTW, take a look at the `before_deactivate_plugin` function, do you prefer to move back there the `empty ( $this->config )` check?